gdnative-core/src/nativescript/init.rs

```diff
@@ -271,13 +282,21 @@ impl<C: NativeClass> ClassBuilder<C> {
     }
 
     #[inline]
-    pub fn add_method(&self, name: &str, method: ScriptMethodFn) {
+    pub fn add_method(&self, name: &str, method: ScriptMethodFn, rpc: &str) {
+        let rpc_mode = match rpc {
+            "remote" => RpcMode::Remote,
+            "remotesync" => RpcMode::RemoteSync,
+            "master" => RpcMode::Master,
+            "puppet" => RpcMode::Puppet,
+            "puppetsync" => RpcMode::PuppetSync,
+            "mastersync" => RpcMode::MasterSync,
+            _ => RpcMode::Disabled,
+        };
+
```

Comment:
- The use cases for RPC modes are uncommon enough that, that they don't warrant inclusion in the main interface (a breaking change) here. Maybe add a new method called `add_method_with_rpc_mode` instead?
- `rpc` should just have the `RpcMode` type.

Reply:
Will do. I'm rather new to GdNative, Rust, and Godot in general so I'm trying to soak up as much knowledge as possible so if it's not too much of a bother could you explain how this is a breaking change? I was able to stand up a godot project and use RPC modes successfully.

Reply:
This is a breaking change because it adds a new parameter, `rpc`, to a method in the public interface, `ClassBuilder::add_method`. Code that call `add_method` directly (not through the `#[methods]` macro) will fail to compile after this change.

Reply:
That makes sense thanks for the explanation!